src/System.Management.Automation/engine/AsyncByteStreamDrainer.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System.Buffers;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Management.Automation;
+
+/// <summary>
+/// Represents the transfer of bytes from one <see cref="Stream" /> to another
+/// asynchronously.
+/// </summary>
+internal sealed class AsyncByteStreamDrainer : IDisposable
+{
+    private const int DefaultBufferSize = 1024;
```

Comment:
I went with the default that .NET uses when you call `Process.BeginOutputReadLine`. Note that this isn't the underlying pipe's buffer size, so it's likely that it will still use the OS's default.